python/ray/data/dataset.py

```diff
@@ -4027,6 +4022,22 @@ def write_parquet(
                 "ignore", "append". Defaults to "append".
                 NOTE: This method isn't atomic. "Overwrite" first deletes all the data
                 before writing to `path`.
+
+        .. note::
+
+            When using `arrow_parquet_args` or `arrow_parquet_args_fn` to pass extra
+            options to pyarrow, there are some special cases:
+
+            - `partitioning_flavor`: if it's not provided, default is "hive" in Ray Data.
+              Otherwise, it follows pyarrow's behavior: None for pyarrow's DirectoryPartitioning,
+              "hive" for HivePartitioning, and "filename" for FilenamePartitioning.
+              See `pyarrow.dataset.partitioning` <https://arrow.apache.org/docs/python/generated/pyarrow.dataset.partitioning.html>_.
+            - `row_group_size`: if it's provided, it will be passed to
+              `pyarrow.parquet.ParquetWriter.write_table() <https:/\
+                  /arrow.apache.org/docs/python/generated/pyarrow\
+                      .parquet.ParquetWriter.html\
+                      #pyarrow.parquet.ParquetWriter.write_table>`_.
+
```

Comment:
Nit: Present tense for consistency with our technical writing style
```suggestion
            - `row_group_size`: if it's provided, it's passed to
              `pyarrow.parquet.ParquetWriter.write_table() <https:/\
                  /arrow.apache.org/docs/python/generated/pyarrow\
                      .parquet.ParquetWriter.html\
                      #pyarrow.parquet.ParquetWriter.write_table>`_.

```